tests/client/test_async_client.py

```diff
@@ -166,3 +166,59 @@ async def test_100_continue(server):
 
     assert response.status_code == 200
     assert response.content == data
+
+
+@pytest.mark.usefixtures("async_environment")
```

Comment:
What do we think of putting these tests (and those for the sync client) in a dedicated `tests/client/test_event_hooks.py` file?